src/ir/module-utils.h

```diff
@@ -399,6 +393,65 @@ template<typename T> struct CallGraphPropertyAnalysis {
   }
 };
 
+// Helper function for collecting the type signature used in a module
+//
+// Used when emitting or printing a module to give signatures canonical
+// indices. Signatures are sorted in order of decreasing frequency to minize the
+// size of their collective encoding. Both a vector mapping indices to
+// signatures and a map mapping signatures to indices are produced.
+inline void
```

Comment:
For the bad reason that it is in a header, so the linker would yell at us if it were not inline because it is included into multiple compilation units. This is a terrible reason, but I feel that adding a new .cpp file for the implementations in this file should be a separate PR.

Reply:
Oh I see. Yeah that should be a separate PR if we decide to do that, but I don't think that's problem only for this file either (we have many header implementations throughout the codebase), so I'm not suggesting you do that or anything 😂